components/MainDonationForm.js

```diff
@@ -13,19 +13,21 @@ export default function MainDonationForm() {
       return;
     }
 
-    // Create an array of all selected checkboxes
+    // Create an array to hold the selected causes
     // selectedCauses becomes the string used on the Stripe checkout page
     let selectedCauses = [];
-    for (let i = 0; i < causes.length; i++) {
-      if (causes[i].isChecked) {
-        selectedCauses.push(' ' + causes[i].text);
-      }
-    }
 
     // If the "Anyone" toggle is on, or if all causes are selected,
     // summarize all causes as "Anyone in Need"
```

Comment:
How does this change corresponds to:
- if all are selected then `selectedCauses` should contain only `' Anyone in Need'`?

Reply:
By placing the for loop into an else statement, we prevent it from running and assign the array manually. Since `isAnyoneInNeedToggled` is determined by the state of the `Anyone in need` switch, we know that if it is true, we don't need to loop over the other boxes, only to set the array value to ` Anyone in Need`. 

As opposed to the previous implementation which would iterate over all checkboxes first, and only then do the anyone check.

Reply:
OK but what about scenario:
- uncheck the `Anyone in need` and leave all checkboxes checked

Previously it was handled as `' Anyone in Need'` - is this still working the same or am I missing something?

Other thing that I'm wondering (that is not related with your changes) is why in `handleClick` we got twice `setIsAnyoneInNeedToggled(!isAnyoneInNeedToggled);` 🙂  ?

Reply:
Ah yeah good catch, I didn't think about that sort if scenario, to avoid that it could be done that unchecking all deselects all boxes, however I can see how that may be not be desired from the charity perspective.

> Other thing that I'm wondering (that is not related with your changes) is why in handleClick we got twice setIsAnyoneInNeedToggled(!isAnyoneInNeedToggled); slightly_smiling_face ?
Honestly no idea.

I'll undo my changes here as really the current behaviour is good enough. It's me over-complicating a non-issue haha!